thunder/executors/nvfuserex_impl.py

```diff
@@ -2209,6 +2209,12 @@ def _linear_check(a: TensorProxy, b: TensorProxy, bias: TensorProxy | None) -> b
     if bias is not None and not is_supported_tensor(bias):
         return False
 
+    # nvFuser supports only fp16 and bf16 inputs. Only checking the first tensor
+    # dtype, as all tensors should have the same dtype which is checked by
+    # linear_meta.
+    if a.dtype not in (dtypes.float16, dtypes.bfloat16):
+        return False
```

Comment:
Wasn't aware there's such restriction.

cc'ing @Priya2698 is this an intended behavior? I thought we are just dispatching to aten now and don't see why it's rejecting full precision.

Reply:
At the moment, we do have that restriction since we use `fusedMultiplySum` and `MmaOp` that only accept bf16/fp16.

Reply:
This restriction will although be lifted with the new IR nodes.

Reply:
Got'ya.

Thanks @IvanYashchuk for catching this. @Priya2698 remember to do a version bump when that restriction is lifted, so we can have a proper behavior inserted here.